src/monmove.cpp

```diff
@@ -1442,3 +1572,31 @@ int monster::turns_to_reach( int x, int y )
 
     return int( turns + .9 ); // Halve (to get turns) and round up
 }
+
+std::pair<item *, tripoint> monster::get_most_desired_item_in_radius( int radius,
+        std::function<float( const item & )> calc_desirability ) const
+{
+    float max_desirability = -1;
+    tripoint desired_item_pos;
+    item *desired_item = nullptr;
+    for( auto &p : g->m.points_in_radius( pos(), radius ) ) {
+        if( g->m.sees_some_items( p, *this ) ) {
+            auto items = g->m.i_at( p );
+            for( auto &itm : items ) {
+                float des = calc_desirability( itm );
+                if( des > max_desirability ) {
+                    desired_item = &itm;
+                    desired_item_pos = p;
+                }
+            }
+        }
+    }
+    return std::make_pair( desired_item, desired_item_pos );
```

Comment:
You probably don't need to return a `pair` here. `desired_item` is never used.

Reply:
Ah — I used it originally but ended up not doing so, but in retrospect it may be useful to future users of this function to have it available. I will, however, change this to an `item_location` rather than a pair, as I didn't know the former existed at the time and it fits exactly this function.